### PR TITLE
Use uid instead of username and return data in JSON body

### LINKED ISSUE #24: Update Cloud Functions to work with uid values instead of usernames
After doing research on writing security rules, it makes more sense to base things off of a uid instead of a username (I didn't even think about uids existing until playing with the auth feature).

There are a number of functions where the public API accepts a `player` parameter which was the username. They should accept a uid instead. Often, we find a document by doing a search for the username - we should be able to simplify this by using the uid as the document ID when possible.

The SDK methods also need to be updated to reflect this change.

### LINKED ISSUE #50: Update Cloud Functions to return data in a JSON body
As part of working on #20, I realized I only return data in headers, for which I did not really have a good reason. It would make sense to return the data in the body instead or in addition.

Part of resolving this issue requires that we decide on a schema for the response format.

## Requirements
- All existing Cloud Functions return data in a JSON body
- A schema is documented in some way (ideally as types in the code)
- Update the tests to account for the new response format

Currently, the rest of the project has not been wired in to the functions, so there should be no breakage.

### PR DESCRIPTION
**Pull request by eeeeeric**
Resolves #24
Resolves #50

Writing those tests paid off again because I definitely broke stuff on the way...

### CODE REVIEW

File: firebase/functions/src/util.ts [hunk 1]
```diff
@@ -46,32 +27,24 @@ export const getUser = async function(username: string): Promise<Firestore.Docum
...
 export const getDeck = async function(username: string, deckId: string): Promise<Firestore.DocumentSnapshot> {
     const userDoc = await getUser(username);
-    let deckDoc;
-    try {
-        deckDoc = await userDoc.ref.collection('Decks').doc(deckId).get();
-    } catch (e) {
-        console.error('An error occured reading from Firestore', e);
-        throw createError(500);
-    }
+    const deckDoc = await userDoc.ref.collection('Decks').doc(deckId).get();
```
Comment (hrecker) [inline firebase/functions/src/util.ts]: You could use the other format here to simplify a little bit, like the SDK does (`Decks/deckId`).

Comment (hrecker) [reply firebase/functions/src/util.ts]: (There are supposed to be backticks around that)

**Comment by hrecker** [comment]
Just had one minor comment.

**Comment by acmorales** [comment]
Looks good to me! I like the Response and Error classes

File: firebase/functions/src/util.ts (hunk 1 above)
Comment (acmorales) [reply firebase/functions/src/util.ts]:
```
`Decks/${deckId}`
```

Comment (eeeeeric) [reply firebase/functions/src/util.ts]: It doesn't seem like `DocumentReference` exposes a `doc` method that I can use like that. https://firebase.google.com/docs/reference/js/firebase.firestore.DocumentReference

**Comment by eeeeeric** [comment]
Thanks y'all.

**Pull request merged**